internal/pkg/v2/infrastructure/redis/event.go

```diff
@@ -131,6 +135,101 @@ func deleteEventById(conn redis.Conn, id string) (edgeXerr errors.EdgeX) {
+	for {
+		select {
+		case eventIds, ok := <-deleteEventsByIdChannel:
+			if ok {
+				c.loggingClient.Debug(fmt.Sprintf("Prepare to delete %v reading data", len(eventIds)))
+				startTime := time.Now()
+				conn := c.Pool.Get()
+				err := deleteEventsByIds(conn, eventIds, c.BatchSize)
+				if err != nil {
+					c.loggingClient.Error(fmt.Sprintf("Deleted events failed.  Err: %s", err.Error()))
+				} else {
+					c.loggingClient.Debug(fmt.Sprintf("Deleted events successfully. elapsed time: %s", time.Since(startTime)))
+				}
+				conn.Close()
+			}
+		}
+	}
+}
+
+// deleteEventsByIds deletes all readings with given event Ids
+func deleteEventsByIds(conn redis.Conn, eventIds []string, batchSize int) (edgeXerr errors.EdgeX) {
+	var events [][]byte
+	//start a transaction to get all events
+	events, edgeXerr = getObjectsByIds(conn, common.ConvertStringsToInterfaces(eventIds))
+	if edgeXerr != nil {
+		return edgeXerr
+	}
+
+	// iterate each events for deletion in batch
+	queriesInQueue := 0
+	e := models.Event{}
+	_ = conn.Send(MULTI)
+	for i, event := range events {
+		err := json.Unmarshal(event, &e)
+		if err != nil {
+			return errors.NewCommonEdgeX(errors.KindContractInvalid, "unable to marshal event", err)
+		}
+		storedKey := eventStoredKey(e.Id)
+		_ = conn.Send(UNLINK, storedKey)
+		_ = conn.Send(UNLINK, fmt.Sprintf("%s:%s", EventsCollectionReadings, e.Id))
+		_ = conn.Send(ZREM, EventsCollection, storedKey)
+		_ = conn.Send(ZREM, EventsCollectionCreated, storedKey)
+		_ = conn.Send(ZREM, EventsCollectionPushed, storedKey)
+		_ = conn.Send(ZREM, fmt.Sprintf("%s:%s", EventsCollectionDeviceName, e.DeviceName), storedKey)
+		queriesInQueue++
+
+		if queriesInQueue >= batchSize {
+			_, err = conn.Do(EXEC)
+			queriesInQueue = 0
+			if err != nil {
+				return errors.NewCommonEdgeX(errors.KindDatabaseError, "unable to execute batch event deletion", err)
+			}
+			// rerun another transaction when event iteration is not finished
+			if i < len(events)-1 {
+				_ = conn.Send(MULTI)
+			}
+		}
+	}
+
+	if queriesInQueue > 0 {
+		_, err := conn.Do("EXEC")
+		if err != nil {
+			return errors.NewCommonEdgeX(errors.KindDatabaseError, "unable to execute batch event deletion", err)
+		}
+	}
+	return nil
+}
+
+// delete all readings and events
+func deleteAllEvents(conn redis.Conn) errors.EdgeX {
+	err := deleteRecordsByKey(conn, ReadingsCollection)
+	if err != nil {
+		return errors.NewCommonEdgeX(errors.KindDatabaseError, "scrub all readings failed", err)
+	}
+	err = deleteRecordsByKey(conn, EventsCollection)
+	if err != nil {
+		return errors.NewCommonEdgeX(errors.KindDatabaseError, "scrub all events failed", err)
+	}
+	return nil
+}
+
+func deleteRecordsByKey(conn redis.Conn, key string) error {
+	_ = conn.Send(MULTI)
+	s := redisScripts[unlinkZSETMembers]
+	_ = s.Send(conn, key)
+	s = redisScripts[unlinkCollection]
+	_ = s.Send(conn, key)
+	_, err := conn.Do(EXEC)
+	return err
+}
```

Comment:
Is this still needed? I.e. do we have a `Purge` feature?

Reply:
We don't define a purge feature at this moment, so these two functions could be removed.

Reply:
Please remove since not in plan.